corehq/apps/locations/templates/locations/manage/locations.html

```diff
@@ -188,14 +188,23 @@
 <div class="row">
     <div class="col-sm-12">
         <div class="btn-toolbar">
-            <a class="btn btn-primary"
+            <a class="btn btn-info"
+               {% if not has_location_types or not can_edit_any_location %}
```

Comment:
I agree that it's probably fine to leave as-is.

Unfortunately we don't really have a cohesive system for location-based permissions, which is something I've talked with product about.  In general though, if access is restricted by locations, then only domain admins can make changes to locations.  I don't think we have a policy on viewing locations.